http/src/main/resources/init-resources/azure_vm_init_script.sh

```diff
@@ -78,20 +70,22 @@ else
   ## Failure to use partprobe can cause the blkid or lslbk commands to not return the UUID for the new filesystem immediately.
   sudo partprobe "${DISK_DEVICE_PATH}1"
   ## Mount the PD partition to the working directory
-  sudo mount -t ext4 "${DISK_DEVICE_PATH}1" ${WORK_DIRECTORY}
+  sudo mount -t ext4 "${DISK_DEVICE_PATH}1" ${PD_DIRECTORY}
   echo "successful mount"
 fi
 
 ## Add the PD UUID to fstab to ensure that the drive is remounted automatically after a reboot
 OUTPUT="$(lsblk -no UUID --paths "${DISK_DEVICE_PATH}1")"
-echo "UUID="$OUTPUT"    ${WORK_DIRECTORY}    ext4    defaults    0    1" | sudo tee -a /etc/fstab
+echo "UUID="$OUTPUT"    ${PD_DIRECTORY}    ext4    defaults    0    1" | sudo tee -a /etc/fstab
 echo "successful write of PD UUID to fstab"
 
-## Change ownership of the mounted drive to the user
-sudo chown -R $VM_JUP_USER:$VM_JUP_USER ${WORK_DIRECTORY}
+## Make sure that both the jupyter and welder users have access to the persistent disk on the VM
+## This needs to happen before we start up containers
+sudo chmod a+rwx ${PD_DIRECTORY}
 
 
-# Read script arguments
+##### READ SCRIPT ARGUMENT #####
+# These are passed in setupCreateVmCreateMessage in the AzurePubsub Handler
```

Comment:
nit, but would it make more sense to check the arguments at the beginning of the script?

Reply:
Happy to move it at the top :)